tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/matter/createMatter.controller.js

```diff
@@ -3391,6 +3497,83 @@
             $rootScope.$on('disableOverlay', function (event, data) {
                 cm.popupContainerBackground = "hide";               
             });
+
+            cm.inputs = [];
+            function getAdditionalMatterProperties(data) {
+                var additionalMatterPropSettingName = configs.taxonomy.matterProvisionExtraPropertiesContentType;
+                if (data[additionalMatterPropSettingName] && data[additionalMatterPropSettingName] != "") {
+                    cm.configurableSection = true;
+                    if (cm.configurableSection) {
+                        cm.createContent.Tab4Header = "Provide more inputs";
+                        cm.createContent.Tab4HeaderTitle = "";
+                        cm.matterAdditionalFieldsContentTypeName = "";
+                        cm.createContent.TabNumber = 4;
+                        if (cm.clientUrl == "") {
+                            cm.clientUrl = configs.global.repositoryUrl;
+                        }
+                        var optionsForGetmatterprovisionextraproperties = {
+                            Client: {
+                                Url: cm.clientUrl
+                            },
+                            MatterExtraProperties: {
+                                ContentTypeName: data[additionalMatterPropSettingName]
+                            }
+                        }
+                        getmatterprovisionextraproperties(optionsForGetmatterprovisionextraproperties, function (result) {
+                            console.log(result);
+                            cm.inputs = result.Fields;
+                            var z = 0;
+                            for (var i = 1; i <= cm.inputs.length; i++) {
+                                var order = (i % 2 == 0) ? 2 : 1;
+                                cm.inputs[z].columnPosition = order;
+                                z++;
+                            }
+                            console.log(cm.inputs);
+                        });
+                    }
+
+                } else {
+                    cm.configurableSection = false;
+                    cm.createContent.TabNumber = 3;
+                    cm.matterAdditionalFieldsContentTypeName = "";
+                    cm.inputs = [];
+                }
+            }
+
+            function getAdditionalMatterPropertiesFieldsData() {
+                var Fields = [];

```

Comment:
To retrieve all fields data on click create and notify button while matter creation.